ingestion-beam/bin/generate-bq-schemas

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# Generate BigQuery schemas from a schema archive fetched by
+# `bin/download=schemas`. This script will write to a flat directory called
+# `bq-schemas` containing BQ schemas that have been injected with partitioning
+# and metadata schemas.
+
+
+set -e
+
+cd "$(dirname "$0")/.." || exit
+
+if [ ! -x  "$(command -v jsonschema-transpiler)" ]; then
+    echo "jsonschema-transpiler is not installed"
+    echo "Run 'cargo install --git https://github.com/acmiyaguchi/jsonschema-transpiler.git --branch dev'"
+    exit 1
+fi
+
+if [ ! -x  "$(command -v jq)" ]; then
+    echo "jq is not installed"
+    exit 1
+fi
+
+if [[ ! -f schemas.tar.gz ]]; then
+    echo "Run 'bin/download-schemas'"
+    exit 1
+fi
+
+function init_working_directory() {
+    # Create a temporary directory for working. The current directory is stored
+    # into $rootdir and the working directory is stored into $workdir. The
+    # working directory will be removed on exit.
+    rootdir=$(pwd)
+    workdir=$(mktemp -d -t tmp.XXXXXXXXXX)
+    function cleanup {
+        rm -rf "$workdir"
+        echo "Running cleanup!"
+    }
+    trap cleanup EXIT
+    cd "$workdir" || exit
+}
+
+total=0
+failed=0
+function generate_bq_schema() {
+    # From a relative path to a jsonschema, transpile an bigquery schema in a
+    # flat directory. This function requires the following global variables:
+    # $total, $failed
+    local src=$1
+    local root=$2
+    
+    local namespace
+    local name
+    # replace // with / and get the proper namespace
+    namespace=$(echo "$src" | sed 's/\/\//\//g' | cut -d/ -f3)
+    name="$namespace.$(basename "$src" .schema.json).bigquery.json"
+    local dst="$root/$name"
+    
+    # Create the folder if not exists
+    mkdir -p "$root"
+
+    if ! jsonschema-transpiler --type bigquery "$src" > "$dst" 2> /dev/null; then
+        echo "Unable to convert $(basename "$src")"
+        rm "$dst"
+        ((failed++))
+    fi
+    tmp="$dst.tmp"
+    
+    # remove top-level record and insert metadata and timestamp
+    submission='{"name": "submission_timestamp", "type": "TIMESTAMP", "mode": "REQUIRED"}'
+    metadata=$(cat << EOM
+{
+    "name": "metadata", 
+    "type": "RECORD",
+    "fields": [
+        {"name": "key", "type": "STRING", "mode": "REQUIRED"},
+        {"name": "value", "type": "STRING", "mode": "REQUIRED"}
+    ],
+    "mode": "REPEATED"
+}
```

Comment:
Instead of adding metadata here, we should merge in the appropriate metadata json schema as incubating in https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/312 ; I expect to have that PR complete and merged on Monday.

Reply:
On that note, to what extent is this script redundant with the script that @fbertsch is working on for merging metadata schemas in with others. I don't want to fragment the logic with multiple implementations. Would it be possible to document in this repo how to pull down and use the other script for creating and updating bq schemas?

Reply:
`update-bq-table` should be treated similarly to `download-schemas`. Neither script should be used during operation because the resources are managed elsewhere. I see this script as a way to set up a minimal staging environment for beam-ingestion for development and validation. I left the metadata as a key-value pair to avoid having to implement the per-namespace logic. 

The tables here should be sandboxed and (generally) short-lived.

Reply:
That makes some more sense. I'm okay with this as long as there's some more documentation at the top of the script to make its intended usage more clear.

Reply:
Also, do you think we should document this workflow in the ingestion-beam README?

Reply:
I'll write something short under `docs/` and point to it from the README.